addons/website_sale/controllers/main.py

```diff
@@ -1003,3 +1003,29 @@ def country_infos(self, country, mode, **kw):
             states=[(st.id, st.name, st.code) for st in country.get_website_sale_states(mode=mode)],
             phone_code=country.phone_code
         )
+
+    @http.route("/shop/cart/<token>", type='http', auth="public", website=True)
+    def access_abandoned_cart(self, token=None, **post):
+        """
+        Token will be pass from revival email template and it will redirect to the selected cart.
+        """
+        order = request.env['sale.order'].sudo().search([('access_token', '=', token)], limit=1)
+        if not order:
+            return request.render('website.404')
+        if order.state != 'draft': # check that cart has been procced or not.
+            response = self.cart()
+            response.qcontext.update({'proceed': 1})
+            return response
+        if post.get('new'):  # redirect to previous cart
+            request.session['sale_order_id'] = order.id
+            return request.redirect("/shop/cart")
+        if post.get('merge'): # merge previous cart into current cart
+            order.order_line.write({'order_id': request.session['sale_order_id']})
```

Comment:
As the customer already filled (at least with essential informations) the partner form in his previous cart, it would be great to get the form pre-filled when he will eventually proceed the cart.